mrjob/py2.py

```diff
@@ -0,0 +1,170 @@
+
+Use the ``StringIO`` from this module to deal with strings (it's
+``StringIO.StringIO`` in Python 2 and ``io.StringIO`` in Python 3).
+
+Please use ``%`` for format strings and not ``format()``, which is much more
+picky about mixing unicode and bytes.
+
+This module provides a ``basestring`` type so you can check if something
+is a "string".
+
+We don't provide a ``unicode`` constant:
+
+- Use ``not isinstance(..., bytes)`` to check if a string is Unicode
+- To convert ``bytes`` to ``unicode``, use ``.decode('utf_8')`.
+- Python 3.3+ has ``u''`` literals; please use sparingly
+
+If you need to convert bytes of unknown encoding to a "string" (e.g.
+to ``print()`` or log them), you can use ``to_string()``.
+
+Iterables
+---------
+
+Using ``.iteritems()`` or ``.itervalues()`` in Python 2 to iterate over a
+dictionary when you don't need a list is best practice, but it's also (in most
+cases) an over-optimization. We'd prefer clean code; just use ``.items()``
+and ``.values()``.
+
+If you *do* every need that extra efficiency ``for k in some_dict`` does not
+create a list in either version of Python, and you for performance-critical
+code, it's fine to write custom code for each Python version (use ``IN_PY2``
+from this module to check which version you're in).
+
+Same goes for ``xrange``; just use ``range`` (this module provides ``xrange``
+solely to support ``ReprProtocol``).
+
+Miscellany
+----------
+
+We provide a ``long`` type (aliased to ``int`` on Python 3) so you can
+check if something is an integer: ``isinstance(..., (int, long))``.
+
+Any standard library function that deals with URLs (e.g. ``urlparse()``) should
+probably be imported from this module.
+
+You *usually* want to do ``from __future__ import print_function`` in modules
+where you use ``print()``. ``print(...)`` works fine, but
+``print(..., file=...)`` doesn't, and ``print()`` prints ``()`` on Python 2.
+
+You shouldn't need any other ``__future__`` imports.
+"""
+import sys
+
+# use this to check if we're in Python 2
+IN_PY2 = (sys.version_info[0] == 2)
+
+# ``basestring``, for ``isinstance(..., basestring)``
+if IN_PY2:
+    basestring = basestring
+else:
+    basestring = str
+basestring
+
+# ``long``, for ``isinstance(..., (int, long))``
+if IN_PY2:
+    long = long
+else:
+    long = int
+long
+
+# ``StringIO``. Useful for mocking out ``sys.stdout``, etc.
+if IN_PY2:
+    from StringIO import StringIO
+else:
+    from io import StringIO
+StringIO  # quiet, pyflakes
+
+# urlparse() (in most cases you should use ``mrjob.parse.urlparse()``)
+if IN_PY2:
+    from urllib import quote
+    from urllib import unquote
+    from urllib2 import urlopen
+    from urlparse import urlparse
+    from urlparse import ParseResult
+else:
+    from urllib.parse import quote
+    from urllib.parse import unquote
+    from urllib.request import urlopen
+    from urllib.parse import urlparse
+    from urllib.parse import ParseResult
+quote
+unquote
+urlopen
+urlparse
+ParseResult
```

Comment:
Err, that is, I think it's a good idea to match the names `six` uses rather than making up my own. But it's even better to just match the names in the Python standard library.